vms/platformvm/vm.go

```diff
@@ -525,9 +525,11 @@ func (vm *VM) GetValidatorSet(ctx context.Context, height uint64, subnetID ids.I
 			return nil, fmt.Errorf("%w: %s", errMissingValidator, vdr.NodeID)
 		}
 		vdrSet[vdr.NodeID] = &validators.GetValidatorOutput{
-			NodeID:    vdr.NodeID,
-			PublicKey: primaryVdr.PublicKey,
-			Weight:    vdr.Weight,
+			NodeID: vdr.NodeID,
+			PublicKey: validators.PublicKey{
```

Comment:
We're probably going to want to do a followup here where we keep the serialized bytes of the keys in memory as well. Will enable us to avoid re-serializing on every call to `GetValidatorSet`

Reply:
Created a followup ticket - https://github.com/ava-labs/avalanchego/issues/1508